apps/meteor/app/api/server/v1/moderation.ts

```diff
@@ -0,0 +1,240 @@
+
+			const uniqueMessages: ReportMessage[] = [];
+			const visited = new Set<string>();
+			for (const report of reports) {
+				if (visited.has(report.message._id)) {
+					continue;
+				}
+				visited.add(report.message._id);
+				uniqueMessages.push(report);
+			}
+
+			return API.v1.success({
+				messages: uniqueMessages,
+				count: reports.length,
+				total,
+				offset,
+			});
+		},
+	},
+);
+
+API.v1.addRoute(
+	'moderation.user.deleteMessageHistory',
+	{
+		authRequired: true,
+		validateParams: isModerationDeleteMsgHistoryParams,
+		permissionsRequired: ['manage-moderation-actions'],
+	},
+	{
+		async post() {
+			// TODO change complicated params
+			const { userId, reasonForHiding } = this.bodyParams;
+
+			const sanitizedReason = reasonForHiding?.trim() ? reasonForHiding : 'No reason provided';
+
+			const { user: moderator } = this;
+
+			const { count = 50, offset = 0 } = await getPaginationItems(this.queryParams);
+
+			const user = await Users.findOneById(userId, { projection: { _id: 1 } });
+			if (!user) {
+				return API.v1.failure('error-invalid-user');
+			}
+
+			const { cursor, totalCount } = ModerationReports.findReportedMessagesByReportedUserId(userId, '', {
+				offset,
+				count,
+				sort: { ts: -1 },
+			});
+
+			const [messages, total] = await Promise.all([cursor.toArray(), totalCount]);
+
+			if (total === 0) {
+				return API.v1.failure('No reported messages found for this user.');
+			}
+
+			await deleteReportedMessages(
+				messages.map((message) => message.message),
+				moderator,
+			);
+
+			await ModerationReports.hideReportsByUserId(userId, this.userId, sanitizedReason, 'DELETE Messages');
+
+			return API.v1.success();
+		},
+	},
+);
+
+API.v1.addRoute(
+	'moderation.markChecked',
+	{
+		authRequired: true,
+		validateParams: isArchiveReportProps,
+		permissionsRequired: ['manage-moderation-actions'],
+	},
+	{
+		async post() {
+			// TODO change complicated camelcases to simple verbs/nouns
+			const { userId, msgId, reasonForHiding, actionTaken } = this.bodyParams;
+
+			if (userId) {
+				const user = await Users.findOneById(userId, { projection: { _id: 1 } });
+				if (!user) {
+					return API.v1.failure('user-not-found');
+				}
+			}
+
+			if (msgId) {
+				const message = await Messages.findOneById(msgId, { projection: { _id: 1 } });
+				if (!message) {
+					return API.v1.failure('error-message-not-found');
+				}
+			}
+
+			const sanitizedReason: string = reasonForHiding?.trim() ? reasonForHiding : 'No reason provided';
+			const action: string = actionTaken ?? 'None';
+
+			const { userId: moderatorId } = this;
+
+			if (userId) {
+				await ModerationReports.hideReportsByUserId(userId, moderatorId, sanitizedReason, action);
+			} else {
+				await ModerationReports.hideReportsByMessageId(msgId as string, moderatorId, sanitizedReason, action);
+			}
+
+			return API.v1.success();
+		},
+	},
+);
+
+API.v1.addRoute(
+	'moderation.reportsByMessage',
```

Comment:
I think we can name it rather simpler - `moderation.reports`. Result will depend on what's passed, messageId or userId(userid not implemented atm).